harness/determined/searcher/search_runner.py

```diff
@@ -49,94 +34,189 @@ def run(
                         operations = [Progress(progress)]
                     else:
-                        raise RuntimeError(f"Unsupported event {e}")
+                        raise RuntimeError(f"Unsupported event {event}")
                     bindings.post_PostSearcherOperations(
                         session,
                         body=bindings.v1PostSearcherOperationsRequest(
-                            experimentId=exp._id,
+                            experimentId=self.search_method.searcher_state.experiment_id,
                             searcherOperations=[op._to_searcher_operation() for op in operations],
-                            triggeredByEvent=e,
+                            triggeredByEvent=event,
                         ),
-                        experimentId=exp._id,
+                        experimentId=experiment_id,
                     )
+
+                    # save state
+                    assert event.id is not None  # TODO change proto to make id mandatory
+                    self.search_method.searcher_state.last_event_id = event.id
+                    self.save_state(experiment_id, event.id)
         except KeyboardInterrupt:
             print("Runner interrupted")
 
+    def save_state(self, experiment_id: int, event_id: int) -> None:
+        pass
+
+    @staticmethod
+    def _searcher_event_as_dict(event: v1SearcherEvent) -> dict:
+        d = {}
+        if event.trialExitedEarly:
+            d["trialExitedEarly"] = event.trialExitedEarly.to_json()
+        if event.validationCompleted:
+            d["validationCompleted"] = event.validationCompleted.to_json()
+        if event.trialProgress:
+            d["trialProgress"] = event.trialProgress.to_json()
+        if event.trialClosed:
+            d["trialClosed"] = event.trialClosed.to_json()
+        if event.trialCreated:
+            d["trialCreated"] = event.trialCreated.to_json()
+        if event.initialOperations:
+            d["initialOperations"] = event.initialOperations.to_json()
+        if event.experimentInactive:
+            d["experimentInactive"] = event.experimentInactive.to_json()
+        d["id"] = event.id
+        return d
+
+
+class LocalSearchRunner(SearchRunner):
+    def __init__(
+        self,
+        search_method: SearchMethod,
+        searcher_dir: Optional[Path] = None,
+    ):
+        super().__init__(search_method)
+        self.state_path = None
+
+        self.searcher_dir = searcher_dir or Path.cwd()
+        if not self.searcher_dir.exists():
+            self.searcher_dir.mkdir(parents=True)
+        elif not self.searcher_dir.is_dir():
+            raise FileExistsError(
+                f"searcher_dir={self.searcher_dir} already exists and is not a directory"
+            )
+
+    def run(
+        self,
+        exp_config: Dict[str, Any],
+        context_dir: Optional[str] = None,
+    ) -> int:
+        """
+        Run custom search without an experiment id
```

Comment:
Are we planning to allow users to resume a specific experiment by providing experiment_id? 
It looks like right now they can edit EXPERIMENT_ID_FILE to resume selected experiment anyway.

Reply:
Not sure I understand the question. Editing EXPERIMENT_ID_FILE is hacky / unsupported.